app/(dashboard)/enroll/[repositoryId]/leaderBoard/components/leaderBoard.tsx

```diff
@@ -0,0 +1,74 @@
+"use client";
+
+import UserProfileSummary from "@/components/ui/userProfileSummary";
+import { TPointTransactionWithUser } from "@/types/pointTransaction";
+import { useCallback, useEffect, useRef, useState } from "react";
+
+import { getPointsOfUsersInRepoByRepositoryIdAction } from "../actions";
```

Comment:
This isn't consistent:

1. you're already passing the user data from the page to the component
2. This should be server-side so you don't need the server action at all

Have a look how we solved it for the open issues :) A lot easier! Please update accordingly.

We're also doing unlimited loading in Formbricks on our Response page. Have a look in the repo there how we do it server side 🤓

This component can be half as long, please refactor.